tests/clvm/coin_store.py

```diff
@@ -79,16 +104,37 @@ def validate_spend_bundle(
 
         return 0
 
-    def update_coin_store_for_spend_bundle(self, spend_bundle: SpendBundle, now: CoinTimestamp, max_cost: int):
+    def update_coin_store_for_spend_bundle(
+        self,
+        spend_bundle: SpendBundle,
+        now: CoinTimestamp,
+        max_cost: int,
+    ):
         err = self.validate_spend_bundle(spend_bundle, now, max_cost)
         if err != 0:
             raise BadSpendBundleError(f"validation failure {err}")
         for spent_coin in spend_bundle.removals():
             coin_name = spent_coin.name()
-            coin_record = self._db[coin_name]
-            self._db[coin_name] = replace(coin_record, spent_block_index=now.height, spent=True)
+            try:
+                coin_record = self._db[coin_name]
+                self._db[coin_name] = replace(
+                    coin_record,
+                    spent_block_index=now.height,
+                    spent=True,
+                )
+            except KeyError:
```

Comment:
I'm pretty sure this will allow non-existent coins in the `CoinStore` to be spent (since it just creates them as already spent) which is not what we want.

Reply:
We should probably do the following:

1. Check that all coins in removals list exist (either in DB or in `additions`, which means they're ephemeral)
2. Add all additions to `._db`
3. Remove all removals from `._db`

Reply:
I changed this in 2434582 but it's still somewhat lacking.  We'll revisit if we decide to do an overhaul of the object.